uqcsbot/scripts/meme.py

```diff
@@ -0,0 +1,308 @@
+I Can Has Cheezburger?: icanhas
+Oh, I'm Sorry, I Thought This Was America: imsorry
+Inigo Montoya: inigo
+The Most Interesting Man in the World: interesting
+Jony Ive Redesigns Things: ive
+Insanity Wolf: iw
+Nothing To Do Here: jetpack
+It's Simple, Kill the Batman: joker
+Probably Not a Good Idea: jw
+Conspiracy Keanu: keanu
+But That's None of My Business: kermit
+Do It Live!: live
+Laughing Lizard: ll
+Member Berries: mb
+Minor Mistake Marvin: mmm
+Shut Up and Take My Money!: money
+One Does Not Simply Walk into Mordor: mordor
+Matrix Morpheus: morpheus
+I Guarantee It: mw
+So I Got That Goin' For Me, Which is Nice: nice
+I Have No Idea What I'm Doing: noidea
+Overly Attached Girlfriend: oag
+That Would Be Great: officespace
+An Older Code Sir, But It Checks Out: older
+Oprah You Get a Car: oprah
+Push it somewhere else Patrick: patrick
+Philosoraptor: philosoraptor
+Unpopular opinion puffin: puffin
+Oh, Is That What We're Going to Do Today?: red
+I Immediately Regret This Decision!: regret
+Pepperidge Farm Remembers: remembers
+Roll Safe: rollsafe
+Sad Joe Biden: sad-biden
+Sad John Boehner: sad-boehner
+Sad George Bush: sad-bush
+Sad Bill Clinton: sad-clinton
+Sad Barack Obama: sad-obama
+Sad Frog / Feels Bad Man: sadfrog
+Salt Bae: saltbae
+Sarcastic Bear: sarcasticbear
+Scumbag Brain: sb
+Sudden Clarity Clarence: scc
+Sealed Fate: sf
+Skeptical Third World Kid: sk
+Super Cool Ski Instructor: ski
+Skeptical Snake: snek
+Seal of Approval: soa
+I Would Be So Happy: sohappy
+So Hot Right Now: sohot
+This is Sparta!: sparta
+Mocking Spongebob: spongebob
+Scumbag Steve: ss
+Baby, You've Got a Stew Going: stew
+Success Kid: success
+10 Guy: tenguy
+The Rent Is Too Damn High: toohigh
+At Least You Tried: tried
+Ugandan Knuckles: ugandanknuck
+What Year Is It?: whatyear
+Winter is coming: winter
+Condescending Wonka: wonka
+X all the Y: xy
+Y'all Got Any More of Them: yallgot
+Xzibit Yo Dawg: yodawg
+Y U NO Guy: yuno
+"""
+
+# TODO: Would be really simple to add custom UQCS memes
+
+
+@bot.on_command("meme")
+@loading_status
+def handle_meme(command: Command):
+    """
+    !meme <meme name> "<top text>" "<bottom text>"
+
+    For a full list of names: !meme names
+    """
+    channel = command.channel_id
+
+    if not command.has_arg():
+        bot.post_message(channel, "Please run !help meme for usage")
+        return
+
+    name = command.arg.split()[0].lower()
+    if name not in VALID_NAMES:
+        bot.post_message(channel, "The meme name is invalid. Try !meme names to get a list of all valid names")
+        return
+
+    args = get_meme_arguments(command.arg)
+
+    if name == "names":
+        send_meme_names(command)
+        return
+    elif len(args) == 2:
+        top, bottom = args
+    else:
+        bot.post_message(channel, "You supplied the wrong number of args. Please run !help meme")
+        return
+
+    # Make an attachment linking to image
+    image_url = API_URL + f"{name}/{top}/{bottom}.jpg"
```

Comment:
Actually, can we make sure to escape the name, top, and bottom args here so we don't get any tricky injection going on

Reply:
Sorry, not quite sure what you mean. What kind of injection could happen here?

Reply:
for example what if the variable "top" contained a forward slash? you need to escape those special characters. pretty sure you want to use `urllib.quote`

Here is an example:
https://github.com/UQComputingSociety/uqcsbot/blob/6a4db88fbf9916984ba3ceb82092f65f58bee74d/uqcsbot/scripts/acronym.py#L14

Reply:
Okay I can do that. Though any forward slashes should be replaced with a ~s in the get_meme_arguments function